app/bundles/EmailBundle/Form/Type/EmailType.php

```diff
@@ -213,7 +218,24 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ]
         );
 
-        $builder->add('isPublished', YesNoButtonGroupType::class);
+        if (!empty($options['data']) && $options['data'] instanceof Email) {
+            $readonly = !$this->security->hasEntityAccess(
+                'email:emails:publishown',
+                'email:emails:publishother',
+                $options['data']->getCreatedBy()
+            );
+            $data = !$readonly && $options['data']->isPublished(false);
```

Comment:
Why does the state of the Published switch depend on the readonly permission? I would think it should show the state as is in the database but not be able to change it.

Reply:
- If it is readonly - we cannot edit publish status, then we  expect the entity is disabled when we create it. For example If I create user without permission publish own -  I assume they cannot create published enity
- If it's editable - then we expect default status of switch button from default value defined in entity

I think it make sense.

Reply:
I may be confused by naming of that variable and double negatives. Let me rewrite it so my brain could understand it:
```php
            $canPublish = $this->security->hasEntityAccess(
                'email:emails:publishown',
                'email:emails:publishother',
                $options['data']->getCreatedBy()
            );
            $data = $canPublish && $options['data']->getIsPublished();
```
So when the users do not have permission to publish then they will always see that the email is unpublished no matter if it is published or not. Why they cannot see the actual unpublish value but read only so they cannot change it? I think it will mislead users thinking it is unpublished even if it's not.

Wait, will this set only the default value when creating new email or will overwrite also on edit? Looking at the [docs](https://symfony.com/doc/current/reference/forms/types/radio.html#data) it looks like it will overwrite on edit too which I find wrong:

> The data option always overrides the value taken from the domain data (object) when rendering. This means the object value is also overridden when the form edits an already persisted object

Reply:
I've made change.
https://github.com/mautic/mautic/pull/11024/files#diff-6128249c053d16b80e1641a97fb28c2ebe64864136fab01b959aa127f37139c8L216

At the moment it just set to false, If entity is new and readonly.